docs/architecture-decisions/cucumber-gherkin-test-suite.md

```diff
@@ -0,0 +1,79 @@
+---
+status: accepted
+author: @toddbaert
+created: 2025-05-16
+updated: --
+---
+
+# Adoption of Cucumber/Gherkin for `flagd` Testing Suite
+
+This decision document outlines the rationale behind adopting the Cucumber/Gherkin testing framework for the `flagd` project’s testing suite. The goal is to establish a clear, maintainable, and language-agnostic approach for writing integration and behavior-driven tests.
+
+By leveraging Gherkin’s natural language syntax and Cucumber’s mature ecosystem, we aim to improve test clarity and accessibility across teams, enabling both developers and non-developers to contribute to test case development and validation.
+
+## Background
+
+`flagd` is a language-agnostic feature flagging engine that forms a core part of the OpenFeature ecosystem. As an open-source project, it needs robust, readable, and accessible testing frameworks that allow for scalable behavior-driven testing.
+
+Previously, test cases for flagd providers were written in language-specific test frameworks, which created fragmentation and limited contributions from engineers who weren’t familiar with the language in question. Furthermore, the ability to validate consistent feature flag behavior across multiple SDKs and environments became increasingly important as adoption grew, and in-process evaluation was implemented.
+
+To address this, the engineering team investigated frameworks that would enable:
+
+- Behavior-driven development (BDD) to validate consistent flag evaluation behavior.
```

Comment:
The existing scenarios appear to cover more than just flag evaluation behavior - it also includes the scenarios like reconnecting to the sync source. Is this intended?

Reply:
I added a bit about this.